src/api/c++/z3++.h

```diff
@@ -448,14 +506,23 @@ namespace z3 {
     public:
         param_descrs(context& c, Z3_param_descrs d): object(c), m_descrs(d) { Z3_param_descrs_inc_ref(c, d); }
         param_descrs(param_descrs const& o): object(o.ctx()), m_descrs(o.m_descrs) { Z3_param_descrs_inc_ref(ctx(), m_descrs); }
+        param_descrs(param_descrs && o) noexcept : object{std::forward<object>(o)}, m_descrs{o.m_descrs} {
+			o.m_descrs = nullptr;
+		}
         param_descrs& operator=(param_descrs const& o) {
             Z3_param_descrs_inc_ref(o.ctx(), o.m_descrs);
             Z3_param_descrs_dec_ref(ctx(), m_descrs);
             m_descrs = o.m_descrs;
-            m_ctx = o.m_ctx;
```

Comment:
Instead of manually copying over data from parent classes, invoke the parent classes' copy operator. This is less brittle and reduces the number of hassles in case the parent class is ever altered.